src/components/Asset/AssetList.tsx

```diff
@@ -51,11 +54,24 @@ export const AssetList = <T extends Token>({
   assets,
   emptyState,
   className,
+  accountId,
   handleSelectToken,
 }: Props<T>) => {
   if (!assets.length) {
     return emptyState || <EmptyAssetList className={className ?? ""} />
   }
+
+  const { data: balances, mutate: fetchBalances } = useGetNearBalances()
+
+  useEffect(() => {
+    if (accountId) {
+      fetchBalances({
+        tokenList: assets.map((item) => item.token),
+        userAddress: accountId,
+      })
+    }
+  }, [accountId, fetchBalances, assets])
```

Comment:
Use can granularly control how useQuery works, for example, you can use `enabled` to disable running the request:
```js
useQuery({..., enabled: accountId != null })
```